### PR TITLE
loader(dm): register tls config when clean  checkpoint

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve? 
* fix https://github.com/pingcap/ticdc/issues/3482
* fix https://github.com/pingcap/ticdc/issues/3511

### What is changed and how it works?

* set right logger config when init lightning
* register and deregister TLS config when removing checkpoint

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

- None

Side effects

 - Increased code complexity

Related changes

- None

### Release note 

```release-note
Please add a release note.
`None`.
```

### CODE REVIEW

**Comment by Ehco1996** [comment]
seems tls test can pass , but still need to reslove the data race

```
==================
WARNING: DATA RACE
Write at 0x00c006317258 by goroutine 67:
  internal/race.Write()
      /Users/ehco/.gvm/gos/go1.17.3/src/internal/race/race.go:42 +0xec
  sync.(*WaitGroup).Wait()
      /Users/ehco/.gvm/gos/go1.17.3/src/sync/waitgroup.go:128 +0xe0
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).waitCheckpointFinish()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:1826 +0x64
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).cleanCheckpoints()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:1830 +0x38
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).cleanCheckpoints-fm()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:1829 +0x50
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).Run()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:446 +0x19c
  github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).run()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/lightning.go:331 +0xc10
  github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).RunOnce()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/lightning.go:201 +0x12c
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).runLightning()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:163 +0x110
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).restore()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:245 +0xee0
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).Process()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:300 +0x544

Previous read at 0x00c006317258 by goroutine 34:
  internal/race.Read()
      /Users/ehco/.gvm/gos/go1.17.3/src/internal/race/race.go:38 +0x130
  sync.(*WaitGroup).Add()
      /Users/ehco/.gvm/gos/go1.17.3/src/sync/waitgroup.go:71 +0x124
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).listenCheckpointUpdates()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:996 +0x4c

Goroutine 67 (running) created at:
  github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).run()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/subtask.go:250 +0x58c
  github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).fetchResultAndUpdateStage()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/subtask.go:324 +0x9cc

Goroutine 34 (running) created at:
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).restoreSchema()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:790 +0x53c
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).restoreSchema-fm()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:740 +0x50
  github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).Run()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/restore/restore.go:446 +0x19c
  github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).run()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/lightning.go:331 +0xc10
  github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).RunOnce()
      /Users/ehco/.gvm/pkgsets/go1.17.3/global/pkg/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211115203106-b076e193b320/br/pkg/lightning/lightning.go:201 +0x12c
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).runLightning()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:163 +0x110
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).restore()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:245 +0xee0
  github.com/pingcap/ticdc/dm/loader.(*LightningLoader).Process()
      /Users/ehco/Documents/pingcap/ticdc/dm/loader/lightning.go:300 +0x544
==================
```

**Comment by Ehco1996** [comment]
data race in lightning_mode
```
==================
WARNING: DATA RACE
Write at 0x00c0005b43c0 by goroutine 238:
  github.com/pingcap/ticdc/dm/pkg/gtid.(*MySQLGTIDSet).Set()
      /Users/ehco/Documents/pingcap/ticdc/dm/pkg/gtid/gtid.go:125 +0x98
  github.com/pingcap/ticdc/dm/relay.(*Relay).handleEvents()
      /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:709 +0x1680
  github.com/pingcap/ticdc/dm/relay.(*Relay).process()
      /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:310 +0x1614
  github.com/pingcap/ticdc/dm/relay.(*Relay).Process()
      /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:154 +0x94
  github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).run()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:140 +0x1b0
  github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).resumeRelay.func1()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:219 +0x80

Previous read at 0x00c0005b43c0 by goroutine 114:
  github.com/pingcap/ticdc/dm/pkg/gtid.(*MySQLGTIDSet).Clone()
      /Users/ehco/Documents/pingcap/ticdc/dm/pkg/gtid/gtid.go:179 +0x7c
  github.com/pingcap/ticdc/dm/relay.(*LocalMeta).GTID()
      /Users/ehco/Documents/pingcap/ticdc/dm/relay/meta.go:353 +0xe0
  github.com/pingcap/ticdc/dm/relay.(*Relay).Status()
      /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:1029 +0x280
  github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).Status()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:158 +0x190
  github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).checkRelayStatus()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:302 +0x94
  github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).check()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:407 +0xa0
  github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).run()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:215 +0x6e0
  github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).Start.func1()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:176 +0x80

Goroutine 238 (running) created at:
  github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).resumeRelay()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:217 +0x298
  github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).Operate()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:183 +0x1a8
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).operateRelay()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:889 +0xf8
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).operateRelayStage()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:878 +0x16c
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).handleRelayStage()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:842 +0x410
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).observeRelayStage()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:791 +0x1d0
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).EnableRelay.func1()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:356 +0xd0

Goroutine 114 (running) created at:
  github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).Start()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:174 +0x78
  github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).Start()
      /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:135 +0xc4
```

**Comment by Ehco1996** [comment]
> data race in lightning_mode
> 
> ```
> ==================
> WARNING: DATA RACE
> Write at 0x00c0005b43c0 by goroutine 238:
>   github.com/pingcap/ticdc/dm/pkg/gtid.(*MySQLGTIDSet).Set()
>       /Users/ehco/Documents/pingcap/ticdc/dm/pkg/gtid/gtid.go:125 +0x98
>   github.com/pingcap/ticdc/dm/relay.(*Relay).handleEvents()
>       /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:709 +0x1680
>   github.com/pingcap/ticdc/dm/relay.(*Relay).process()
>       /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:310 +0x1614
>   github.com/pingcap/ticdc/dm/relay.(*Relay).Process()
>       /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:154 +0x94
>   github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).run()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:140 +0x1b0
>   github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).resumeRelay.func1()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:219 +0x80
> 
> Previous read at 0x00c0005b43c0 by goroutine 114:
>   github.com/pingcap/ticdc/dm/pkg/gtid.(*MySQLGTIDSet).Clone()
>       /Users/ehco/Documents/pingcap/ticdc/dm/pkg/gtid/gtid.go:179 +0x7c
>   github.com/pingcap/ticdc/dm/relay.(*LocalMeta).GTID()
>       /Users/ehco/Documents/pingcap/ticdc/dm/relay/meta.go:353 +0xe0
>   github.com/pingcap/ticdc/dm/relay.(*Relay).Status()
>       /Users/ehco/Documents/pingcap/ticdc/dm/relay/relay.go:1029 +0x280
>   github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).Status()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:158 +0x190
>   github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).checkRelayStatus()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:302 +0x94
>   github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).check()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:407 +0xa0
>   github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).run()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:215 +0x6e0
>   github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).Start.func1()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:176 +0x80
> 
> Goroutine 238 (running) created at:
>   github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).resumeRelay()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:217 +0x298
>   github.com/pingcap/ticdc/dm/dm/worker.(*realRelayHolder).Operate()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/relay.go:183 +0x1a8
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).operateRelay()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:889 +0xf8
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).operateRelayStage()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:878 +0x16c
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).handleRelayStage()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:842 +0x410
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).observeRelayStage()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:791 +0x1d0
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).EnableRelay.func1()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:356 +0xd0
> 
> Goroutine 114 (running) created at:
>   github.com/pingcap/ticdc/dm/dm/worker.(*realTaskStatusChecker).Start()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/task_checker.go:174 +0x78
>   github.com/pingcap/ticdc/dm/dm/worker.(*SourceWorker).Start()
>       /Users/ehco/Documents/pingcap/ticdc/dm/dm/worker/source_worker.go:135 +0xc4
> ```

wow this raced in relay, https://github.com/pingcap/ticdc/pull/3469 may fix this

File: dm/loader/lightning.go [hunk 1]
```diff
@@ -229,14 +234,18 @@ func (l *LightningLoader) restore(ctx context.Context) error {
...
-		cfg.TiDB.Vars = map[string]string{
-			"time_zone": l.timeZone,
-		}
+		cfg.TiDB.Vars = map[string]string{"time_zone": l.timeZone}
 		if err = cfg.Adjust(ctx); err != nil {
 			return err
 		}
 		err = l.runLightning(ctx, cfg)
 		if err == nil {
+			// lightning will auto deregister tls when task done, so we need to register it again for removing checkpoint
```
Comment (lance6716) [inline dm/loader/lightning.go:243]: we might let lightning not deregister the certificate. If there're more than one subtasks, the certificate may still be deregister between line 245 and line 249

File: dm/loader/lightning.go [hunk 2]
```diff
@@ -98,12 +101,15 @@ func makeGlobalConfig(cfg *config.SubTaskConfig) *lcfg.GlobalConfig {
...
 	lightningCfg.TiDB.PdAddr = cfg.TiDB.PdAddr
+	lightningCfg.TiDB.LogLevel = cfg.LogLevel
 	lightningCfg.TikvImporter.Backend = cfg.TiDB.Backend
 	lightningCfg.PostRestore.Checksum = lcfg.OpLevelOff
 	if cfg.TiDB.Backend == lcfg.BackendLocal {
 		lightningCfg.TikvImporter.SortedKVDir = cfg.Dir
 	}
 	lightningCfg.Mydumper.SourceDir = cfg.Dir
+	lightningCfg.App.Config.Level = cfg.LogLevel
+	lightningCfg.App.Config.File = cfg.LogFile
```
Comment (glorv) [inline dm/loader/lightning.go:112]: We should net let lightning init log another time. https://github.com/pingcap/tidb/pull/29291/ this PR will skip init lightning log so lightning will share the global config with dm.

**Comment by Ehco1996** [comment]
last one is [data race](https://ci2.pingcap.net/blue/organizations/jenkins/dm_ghpr_integration_test/detail/dm_ghpr_integration_test/1337/pipeline/) , will fix in next commit

/run-dm-integration-tests

**Comment by Ehco1996** [comment]
```

[2021-12-07T08:15:21.155Z] ==================

[2021-12-07T08:15:21.155Z] WARNING: DATA RACE

[2021-12-07T08:15:21.155Z] Write at 0x00c006af45d8 by goroutine 149:

[2021-12-07T08:15:21.155Z]   internal/race.Write()

[2021-12-07T08:15:21.155Z]       /usr/local/go/src/internal/race/race.go:41 +0x125

[2021-12-07T08:15:21.155Z]   sync.(*WaitGroup).Wait()

[2021-12-07T08:15:21.155Z]       /usr/local/go/src/sync/waitgroup.go:128 +0x126

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).waitCheckpointFinish()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:1826 +0x94

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).cleanCheckpoints()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:1830 +0x4f

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).cleanCheckpoints-fm()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:1829 +0x6a

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).Run()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:446 +0x24f

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).run()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/lightning.go:330 +0x10e4

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).RunOnce()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/lightning.go:201 +0x184

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).runLightning()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:167 +0x15a

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).restore()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:225 +0x1284

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).Process()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:289 +0x62f

[2021-12-07T08:15:21.155Z] 

[2021-12-07T08:15:21.155Z] Previous read at 0x00c006af45d8 by goroutine 165:

[2021-12-07T08:15:21.155Z]   internal/race.Read()

[2021-12-07T08:15:21.155Z]       /usr/local/go/src/internal/race/race.go:37 +0x206

[2021-12-07T08:15:21.155Z]   sync.(*WaitGroup).Add()

[2021-12-07T08:15:21.155Z]       /usr/local/go/src/sync/waitgroup.go:71 +0x219

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).listenCheckpointUpdates()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:996 +0x6f

[2021-12-07T08:15:21.155Z] 

[2021-12-07T08:15:21.155Z] Goroutine 149 (running) created at:

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).run()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:250 +0x769

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/dm/worker.(*SubTask).fetchResultAndUpdateStage()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/dm/worker/subtask.go:324 +0x1184

[2021-12-07T08:15:21.155Z] 

[2021-12-07T08:15:21.155Z] Goroutine 165 (finished) created at:

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).restoreSchema()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:790 +0x698

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).restoreSchema-fm()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:740 +0x6a

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning/restore.(*Controller).Run()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/restore/restore.go:446 +0x24f

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).run()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/lightning.go:330 +0x10e4

[2021-12-07T08:15:21.155Z]   github.com/pingcap/tidb/br/pkg/lightning.(*Lightning).RunOnce()

[2021-12-07T08:15:21.155Z]       /nfs/cache/mod/github.com/pingcap/tidb@v1.1.0-beta.0.20211207072534-fa7219365f31/br/pkg/lightning/lightning.go:201 +0x184

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).runLightning()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:167 +0x15a

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).restore()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:225 +0x1284

[2021-12-07T08:15:21.155Z]   github.com/pingcap/ticdc/dm/loader.(*LightningLoader).Process()

[2021-12-07T08:15:21.155Z]       /home/jenkins/agent/workspace/dm_ghpr_integration_test/go/src/github.com/pingcap/ticdc/dm/loader/lightning.go:289 +0x62f

[2021-12-07T08:15:21.155Z] ==================
```

File: dm/loader/lightning.go [hunk 3]
```diff
@@ -204,39 +207,30 @@ func (l *LightningLoader) restore(ctx context.Context) error {
...
 			for k, v := range l.cfg.To.Session {
 				cfg.TiDB.Vars[k] = v
 			}
 		}
-
 		cfg.TiDB.StrSQLMode = l.cfg.LoaderConfig.SQLMode
-		cfg.TiDB.Vars = map[string]string{
-			"time_zone": l.timeZone,
-		}
-		if err = cfg.Adjust(ctx); err != nil {
```
Comment (Ehco1996) [inline dm/loader/lightning.go:235]: lightning will auto adjust config in [hehe](https://github.com/pingcap/tidb/blob/d83ee8cfdad434d212d3229c0be24d9d81ef7ab3/br/pkg/lightning/lightning.go#L192), don't need to adjust again

File: dm/loader/lightning.go (hunk 1 above)
Comment (Ehco1996) [reply dm/loader/lightning.go:227]: now lightning allow  register tls config with different name, and it test is passed, ptal 🧡

File: dm/loader/lightning.go (hunk 2 above)
Comment (Ehco1996) [reply dm/loader/lightning.go]: fixed ptal

File: dm/loader/lightning.go (hunk 1 above)
Comment (glorv) [reply dm/loader/lightning.go:227]: CheckpointRemove should be postponed to cleanmeta. I will add this logic in a following PR, so LGTM.

File: dm/loader/lightning.go [hunk 4]
```diff
@@ -299,11 +293,8 @@ func (l *LightningLoader) Process(ctx context.Context, pr chan pb.ProcessResult)
 		isCanceled = true
 	default:
 	}
+	pr <- pb.ProcessResult{IsCanceled: isCanceled, Errors: errs}
```
Comment (lance6716) [inline dm/loader/lightning.go:296]: why move this code? when `pr` is full (though not possible now) the original log can help us debug it.

File: dm/tests/tls/run.sh [hunk 5]
```diff
@@ -73,13 +73,79 @@ function setup_mysql_tls() {
 	echo "add dm_tls_test user done $mysql_data_path"
 }
 
-function test_worker_download_certs_from_master() {
-	cleanup_data tls
+function prepare_test() {
 	cleanup_process
 
+	# clena test dir
```
Comment (lance6716) [inline dm/tests/tls/run.sh:79]:
```suggestion
	# clean test dir
```

File: dm/tests/tls/run.sh [hunk 6]
```diff
@@ -73,13 +73,79 @@ function setup_mysql_tls() {
...
+
+	echo "start task and check stage"
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"start-task $WORK_DIR/dm-task.yaml --remove-meta=true"
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"start-task $WORK_DIR/dm-task-2.yaml --remove-meta=true"
+
+	run_dm_ctl_with_tls_and_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" $cur/conf/ca.pem $cur/conf/dm.pem $cur/conf/dm.key \
+		"query-status test" \
+		"\"result\": true" 2
```
Comment (lance6716) [inline dm/tests/tls/run.sh:135]: check here and below tasks are in sync unit

File: dm/loader/lightning.go (hunk 3 above)
Comment (glorv) [reply dm/loader/lightning.go:235]: hehe 😄

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: dm/loader/lightning.go [hunk 7]
```diff
@@ -345,7 +336,9 @@ func (l *LightningLoader) Resume(ctx context.Context, pr chan pb.ProcessResult)
 		l.logger.Warn("try to resume, but already closed")
 		return
 	}
-	l.core = lightning.New(l.lightningConfig)
+	l.Lock()
```
[the remark this answers is not in the document]
Comment (Ehco1996) [reply dm/loader/lightning.go]: i think resume won't be  concurrently call remove this lock in https://github.com/pingcap/ticdc/pull/3522/commits/fa145d004e003c815e258d02274f4911a5e393c6

**Review by lance6716 — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Pull request merged**